Formula/g/gnupg.rb

```diff
@@ -1,13 +1,13 @@
 class Gnupg < Formula
   desc "GNU Pretty Good Privacy (PGP) package"
   homepage "https://gnupg.org/"
-  url "https://gnupg.org/ftp/gcrypt/gnupg/gnupg-2.4.6.tar.bz2"
-  sha256 "95acfafda7004924a6f5c901677f15ac1bda2754511d973bb4523e8dd840e17a"
+  url "https://gnupg.org/ftp/gcrypt/gnupg/gnupg-2.4.7.tar.bz2"
+  sha256 "7b24706e4da7e0e3b06ca068231027401f238102c41c909631349dcc3b85eb46"
   license "GPL-3.0-or-later"
 
   livecheck do
     url "https://gnupg.org/ftp/gcrypt/gnupg/"
-    regex(/href=.*?gnupg[._-]v?(\d+(?:\.\d+)+)\.t/i)
+    regex(/href=.*?gnupg[._-]v?(2\.[46](?:\.\d+)+)\.t/i)
```

Comment:
I disagree.
At the moment, only 2.6 exists in the plan. See https://gnupg.org/download/index.html#end-of-life
It will take several years before the GnuPG project gets to the successor to 2.6

One can remember Linux version numbering where after 2.0, 2.2, 2.4, 2.6 sequence was replaced with a completely different numbering scheme.

Besides, we patch almost every single "patch" release of GnuPG because build breaks.
The simple don't support macOS as build and run target -- it works mostly because Linux and macOS have more similarities than differences.
So, there would be a lot of occasions when a knowledgeable maintainer would have a chance to fix the livecheck.

Reply:
I think it's fine to stick with 2.4 and 2.6 in the event the version scheme changes in the future. We'll have plenty of time to deal with it.

Reply:
We usually err on the side of false-positives in `livecheck` blocks, at least when unknown versions are concerned. It's a lot easier to skip an update that `livecheck` flagged that we don't want. We also often miss out on new versions due to stale livecheck blocks.

If you don't want to rely on the even-minor version scheme, then we should instead do something like
```suggestion
    regex(/href=.*?gnupg[._-]v?(\d+\.\d*[0-46-9](?:\.\d+)+)\.t/i)
```

Reply:
I disagree and I won't accept this suggestion.
Again, it is specified that there exists a single stable branch (2.4) and another single stable branch is planned (2.6).
https://gnupg.org/download/index.html#end-of-life
The planned branch is several years ahead.
As @p-linnane told, it is fine to stick with these two branches.

Your suggestion, however, excludes versions like 2.5, 2.15, 3.5, 121.45.
This is not the correct regular expression for the task you've imagined.

Reply:
> Your suggestion, however, excludes versions like 2.5, 2.15, 3.5, 121.45.

So does the regex in your patch, which means the following comment also seems to apply to your proposed changes:

> This is not the correct regular expression for the task you've imagined.